docs/guide/creating_a_gem.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: default
+title: Creating a Gem
+---
+
+# Creating a Gem
+
+Start by creating a new Rails engine with the following arguments:
+
+```sh
+rails plugin new shared-view_components \
+  --full \
+  --skip-namespace \
+  --skip-action-mailer \
+  --skip-action-mailbox \
+  --skip-action-text \
+  --skip-active-record \
+  --skip-active-job \
+  --skip-active-storage \
+  --skip-action-cable \
+  --skip-turbolinks \
+  --skip-jbuilder \
+  --skip-gemfile-entry # only use this when adding a gem into an existing Rails app
+```
+
+Update the gemspec. Add the dependency:
+
+```ruby
+spec.add_dependency "view_component", ">= 2.32"
+```
+
+Afterwards, add the require command into `shared/view_components.rb`:
+
+```ruby
+require "view_component/engine"
+```
+
+Let's create our first component:
+
+```sh
+✗ bin/rails g component Shared::Example title --skip-namespace
+      create  app/components/shared/example_component.rb
+      invoke  test_unit
+      create    test/components/shared/example_component_test.rb
+      invoke  erb
+      create    app/components/shared/example_component.html.erb
+```
+
+Update `example_component.html.erb` to the following:
+
+```erb
+<span title="<%= @title %>"><%= content %></span>
+```
+
+Now, let't write a test for the component. Open `test/components/example_component_test.rb`
```

Comment:
What do you think about using headings for each step? 

I could see having a `Testing` heading here.